docs/src/main/asciidoc/spanner.adoc

```diff
@@ -25,51 +29,178 @@ dependencies {
+
+With our Spring Boot autoconfiguration your application is able to inject an instance of `DatabaseClient` and `SpannerOperations`, and when repositories are enabled, all the user defined repositories extending `SpannerRepository` are exposed as beans as well.
 
-=== Google Cloud Spanner
 
-Spring Data Spanner allows you to define entity classes and corresponding repository interfaces:
+=== Mapping POJO domain objects to Spanner tables using annotations
+
+Spring Data Spanner allows you to map POJOs to Spanner tables:
 
 [source,java]
 ----
-@SpannerTable(name = "students")
-public class SimpleStudent {
-	@SpannerPrimaryKeyColumn(keyOrder = 1)
-	String id;
+@Table(name = "trades")
+public class Trade {
+
+	@PrimaryKeyColumn(keyOrder = 2)
+	String action;
+
+	Double price;
 
-	@SpannerColumn(name = "first_name")
-	String name;
+	Double shares;
 
-	@SpannerColumnInnerType(innerType = String.class)
-	List<String> teachers;
+	@PrimaryKeyColumn(keyOrder = 1)
+	String symbol;
 
-	int age;
+	@Column(name = "trader_id")
+	String traderId;
+
+	@ColumnInnerType(innerType = Double.class)
+	List<Double> curve;
 }
 ----
 
-The `@SpannerTable` annotation provides the name of the Spanner table that stores instances of this class, one per row.
-The `@SpannerPrimaryKeyColumn` annotation is required and identifies the properties corresponding to the primary key columns in
+- `@Table` provides the name of the Spanner table that stores instances of this class, one per row.
+- `@PrimaryKeyColumn` annotation is required and identifies the properties corresponding to the primary key columns in
 order starting with 1 and increasing consecutively.
-The `@SpannerColumn` annotation optionally provides a different column name than that of the property.
-The `@SpannerColumnInnerType` annotation is required for List properties due to Java's type erasure.
+- The `@Column` annotation optionally provides a different column name than that of the property.
+- Spanner supports `List` types as fields, the `@ColumnInnerType` annotation is required for `List` properties due to Java's type erasure.
+
+#### SpEL expressions for table names
+
+In some cases you might want the `@Table` table name to be determined dynamically. In this case, you can use https://docs.spring.io/spring/docs/current/spring-framework-reference/core.html#expressions[Spring Expression Language].
+
+For example:
+
+[source, java]
+----
+
+@Table(name = "trades_#{tableNameSuffix}")
+public class Trade {
+	// ...
+}
+----
+
+This table name will be resolved only if `tableNameSuffix` value/bean in the Spring `applicationContext` is defined, and then, for exapmle if it has the value "123", it will be `trades_123`.
+
+=== Template API
+
+`SpannerOperations` and its implementation, `SpannerTemplate` provides the XYZTemplate pattern familiar for Spring developers and as such provides:
+
+ - resource management
+ - one stop shop to Spanner operations with the Spring Data POJO mapping and conversion features
```

Comment:
"one stop shop" should be hyphenated "one-stop-shop". Otherwise it implies we should be interpreting "shop" as a verb.